python/ray/serve/_private/replica.py

```diff
@@ -435,6 +441,27 @@ async def _get_metadata(
         ) -> Tuple[DeploymentConfig, DeploymentVersion]:
             return self.replica.version.deployment_config, self.replica.version
 
+        def _save_cpu_profile_data(self) -> str:
+            """Saves CPU profiling data, if CPU profiling is enabled.
+
+            Logs a warning if CPU profiling is disabled.
+            """
+
+            if self.cpu_profiler is not None:
+                import marshal
+
+                self.cpu_profiler.snapshot_stats()
+                with open(self.cpu_profiler_log, "wb") as f:
+                    marshal.dump(self.cpu_profiler.stats, f)
+                logger.info(f'Saved CPU profile data to file "{self.cpu_profiler_log}"')
+                return self.cpu_profiler_log
+            else:
+                logger.warning(
+                    "Attempted to save CPU profile data, but failed because no "
+                    "CPU profiler was running! Enabled CPU profiling by enabling "
+                    "the RAY_SERVE_ENABLE_CPU_PROFILING env var."
+                )
```

Comment:
this should raise an exception instead of logging a warning (else the task you call just succeeds silently)

same for the other ones

Reply:
Make sense, I updated the statements to raise an exception.